packages/amplify-migration-e2e/src/__tests__/migration_templategen_e2e.test.ts

```diff
@@ -1,56 +1,59 @@
+
+const CATEGORIES_TO_MOVE: RefactorCategory[] = ['auth', 'storage'];
 
 void describe('Templategen E2E tests', () => {
   void describe('Full Migration Templategen Flow', () => {
     let projRoot: string;
     let projName: string;
-    let bucketName: string;
 
     beforeEach(async () => {
       const baseDir = process.env.INIT_CWD ?? process.cwd();
       projRoot = await createNewProjectDir('templategen_e2e_flow_test', path.join(baseDir, '..', '..'));
       projName = `test${generateRandomShortId()}`;
-      bucketName = `testbucket${generateRandomShortId()}`;
     });
 
     afterEach(async () => {
       await cleanupProjects(projRoot, projName);
-      await deleteS3Bucket(bucketName);
     });
 
     void it('should init a project & add auth, function, storage, api with defaults & perform refactor', async () => {
+      // Arrange
       await setupAndPushDefaultGen1Project(projRoot, projName);
-      const { gen1StackName, gen1FunctionName } = await assertDefaultGen1Setup(projRoot);
-      assert(gen1StackName);
-      runCodegenCommand(projRoot);
-      copyFunctionFile(projRoot, 'function', gen1FunctionName);
-      copyGen1Schema(projRoot, projName);
-      npmInstall(projRoot);
-      const gen2StackName = await runGen2SandboxCommand(projRoot, projName);
-      assert(gen2StackName);
-      runExecuteCommand(projRoot, gen1StackName, gen2StackName);
-    });
 
-    void it('should init a project & add auth, function, storage, api with defaults, perform refactor and revert to the original state', async () => {
-      await setupAndPushDefaultGen1Project(projRoot, projName);
-      const { gen1StackName, gen1FunctionName, gen1Region } = await assertDefaultGen1Setup(projRoot);
-      await createS3Bucket(bucketName, gen1Region);
+      // Act
+      const { gen1StackName, gen1FunctionName, envName } = await assertDefaultGen1Setup(projRoot);
       assert(gen1StackName);
       runCodegenCommand(projRoot);
       copyFunctionFile(projRoot, 'function', gen1FunctionName);
-      copyGen1Schema(projRoot, projName);
+      removeErrorThrowsFromFunctionFile(projRoot, 'function', extractFunctionResourceName(gen1FunctionName, envName));
+      updateAmplifyBackendPackagesVersion(projRoot);
       npmInstall(projRoot);
+      // Below env is only needed for CI/CD deployments and is expected to be set by customers for their app
+      // To emulate the migration in sandbox, we set it explicitly.
+      process.env.AMPLIFY_GEN_1_ENV_NAME = envName;
       const gen2StackName = await runGen2SandboxCommand(projRoot, projName);
       assert(gen2StackName);
+
       runExecuteCommand(projRoot, gen1StackName, gen2StackName);
+      await runGen2DeployPostExecute(projRoot, projName, envName, CATEGORIES_TO_MOVE);
+
+      // Assert
+      await assertExecuteCommand(projRoot, CATEGORIES_TO_MOVE);
+
       runRevertCommand(projRoot, gen1StackName, gen2StackName);
```

Comment:
TODO: add assertion post revert once the `GetAtt` removal changes are released in the next tagged release.